client/src/main/java/io/pravega/client/stream/impl/ReaderGroupState.java

```diff
@@ -1228,6 +1229,51 @@ private void write00(ClearCheckpointsBefore object, RevisionDataOutput out) thro
         }
     }
 
+    @Builder
+    @Data
+    @EqualsAndHashCode(callSuper = false)
+    static class RemoveOutstandingCheckpoints extends ReaderGroupStateUpdate {
+
+        /**
+         * @see ReaderGroupState.ReaderGroupStateUpdate#update(ReaderGroupState)
+         */
+        @Override
+        void update(ReaderGroupState state) {
+            state.checkpointState.removeOutstandingCheckpoints();
+        }
+
+        private static class RemoveOutstandingCheckpointsBuilder implements ObjectBuilder<RemoveOutstandingCheckpoints> {
+
+        }
+
+
+        private static class RemoveOutstandingCheckpointsSerializer
+                extends VersionedSerializer.WithBuilder<RemoveOutstandingCheckpoints, RemoveOutstandingCheckpointsBuilder> {
+            @Override
+            protected RemoveOutstandingCheckpointsBuilder newBuilder() {
+                return builder();
+            }
+
+            @Override
+            protected byte getWriteVersion() {
+                return 0;
+            }
+
+            @Override
+            protected void declareVersions() {
+                version(0).revision(0, this::write00, this::read00);
+            }
+
+           private void read00(RevisionDataInput in, RemoveOutstandingCheckpointsBuilder builder) throws IOException {
+               builder.build();
+            }
+
+            private void write00(RemoveOutstandingCheckpoints object, RevisionDataOutput out) throws IOException {
+                out.writeUTF(object.toString());
```

Comment:
Here a string is being written when serializing but it not being read when deserializing. So what's the point in writing it?

Reply:
Have made the above suggested change.